test/jdk/javax/swing/JComboBox/EditableComboBoxPopupPos.java

```diff
@@ -0,0 +1,151 @@
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.BorderFactory;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+public class EditableComboBoxPopupPos {
+    private static Robot robot;
+    private static JFrame frame;
+    private static JPanel panel;
+    private static JComboBox cb1, cb2;
+    private static String lafName, cb1Str, cb2Str;
+    private static Point cb1Point, cb2Point;
+    private static int cb1Width, cb1Height, cb2Width, cb2Height;
+
+    private static final int BUTTON_OFFSET = 8;
+    private static final int POPUP_OFFSET = 6;
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException, AWTException, IOException {
+        robot = new Robot();
+        robot.setAutoDelay(100);
+        robot.setAutoWaitForIdle(true);
+
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+        lafName = laf.getName();
+        SwingUtilities.invokeAndWait(() -> {
+            setLookAndFeel(laf);
+            panel = new JPanel();
+            String[] comboStrings = {"One", "Two", "Three"};
+            cb1 = new JComboBox(comboStrings);
+            cb1.setEditable(true);
+            cb1.setBorder(BorderFactory.createTitledBorder(
+                    "Editable JComboBox"));
+
+            cb2 = new JComboBox(comboStrings);
+            cb2.setEditable(true);
+
+            panel.add(cb1);
+            panel.add(cb2);
+
+            // Change starting selection to check if the position of the
+            // first selection item is in the correct position on screen.
+            cb1.setSelectedIndex(1);
+            cb2.setSelectedIndex(1);
+
+            frame = new JFrame();
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.add(panel);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+            frame.setVisible(true);
+        });
+
+        robot.delay(1000);
+        robot.waitForIdle();
+
+        SwingUtilities.invokeAndWait(() -> {
+            cb1Point = cb1.getLocationOnScreen();
+            cb1Width = cb1.getWidth();
+            cb1Height = cb1.getHeight();
+            cb2Point = cb2.getLocationOnScreen();
+            cb2Width = cb2.getWidth();
+            cb2Height = cb2.getHeight();
+        });
+
+        runTestOnComboBox(cb1Point, cb1Width, cb1Height);
+        runTestOnComboBox(cb2Point, cb2Width, cb2Height);
+
+        SwingUtilities.invokeAndWait(() -> {
+            cb1Str = cb1.getSelectedItem().toString();
+            cb2Str = cb2.getSelectedItem().toString();
+        });
+
+        checkSelection(cb1Str, cb2Str);
+
+        SwingUtilities.invokeAndWait(() -> frame.dispose());
```

Comment:
`dispose()` will not be called if `checkSelection()` will throw an exception.
So you should at least call `checkSelection()` after frame disposal.

But as Alexey already said, more robust way is to call it from finally block, so we end up with a closed frame on any unexpected exception.

Reply:
Sure, I'll add this